alacritty_terminal/src/tty/windows/conpty.rs

```diff
@@ -230,6 +241,35 @@ pub fn new(config: &Options, window_size: WindowSize) -> Option<Pty> {
     Some(Pty::new(conpty, conout, conin, child_watcher))
 }
 
+fn convert_custom_env(custom_env: &HashMap<String, String>) -> Option<Vec<u16>> {
+    let mut result = Vec::new();
+    if custom_env.is_empty() {
+        return None;
+    }
+
+    let mut all_env_keys = HashSet::new();
+    for (custom_key, custom_value) in custom_env {
+        let custom_key = OsStr::new(custom_key);
+        if all_env_keys.insert(custom_key.to_ascii_uppercase()) {
+            result.extend(custom_key.encode_wide());
+            result.push('=' as u16);
+            result.extend(OsStr::new(custom_value).encode_wide());
+            result.push(0);
+        }
+    }
+    for (inherited_key, inherited_value) in std::env::vars_os() {
```

Comment:
I'm also pretty sure that you can merge both iterators, just need a `map` in one and then `chain` the other.

```suggestion
    // Pull the current a process environment after to not overwrite user provided one.
    for (inherited_key, inherited_value) in std::env::vars_os() {
```

Reply:
The problem with that is that `custom_ref` is a reference type, but `std::env::var_os()` is an iterator of owned strings.
So it's not that easy to combine those, but I've extracted the string creation method and documented it, as proposed.